### PR TITLE
Remove deprecated _upgrade API

### LINKED ISSUE #21337: Remove the upgrade API
The upgrade API is no longer useful for adapting an old index for the next major version+1. Instead, the reindex API should be used instead.  

We should remove the upgrade API.

### PR DESCRIPTION
**Pull request by danhermann**
The `_upgrade` API has been superseded by the `_reindex` API and was deprecated in 7.11.0. This commit removes it completely for the 8.0.0 release.

Resolves #21337

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java [hunk 1]
```diff
@@ -181,19 +181,19 @@ public void testTransportReplicationAllShardsTasks() {
     }
 
     public void testTransportBroadcastByNodeTasks() {
-        registerTaskManagerListeners(UpgradeAction.NAME);  // main task
-        registerTaskManagerListeners(UpgradeAction.NAME + "[n]"); // node level tasks
+        registerTaskManagerListeners(ForceMergeAction.NAME);  // main task
+        registerTaskManagerListeners(ForceMergeAction.NAME + "[n]"); // node level tasks
```
Comment (danhermann) [inline server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java:185]: Replaced with a different index action so tasks are still tested

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java [hunk 2]
```diff
@@ -316,36 +312,4 @@ private static boolean maybeUpdateClusterBlock(String[] actualIndices, ClusterBl
         }
         return changed;
     }
-
-
-    public void upgradeIndexSettings(final UpgradeSettingsClusterStateUpdateRequest request,
-                                     final ActionListener<AcknowledgedResponse> listener) {
```
Comment (danhermann) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java:322]: Did not appear to be called outside of the `_upgrade` API.

File: server/src/main/java/org/elasticsearch/index/shard/IndexShard.java [hunk 3]
```diff
@@ -1099,28 +1098,6 @@ public void forceMerge(ForceMergeRequest forceMerge) throws IOException {
             forceMerge.onlyExpungeDeletes(), false, false, forceMerge.forceMergeUUID());
     }
 
-    /**
-     * Upgrades the shard to the current version of Lucene and returns the minimum segment version
-     */
-    public org.apache.lucene.util.Version upgrade(UpgradeRequest upgrade) throws IOException {
```
Comment (danhermann) [inline server/src/main/java/org/elasticsearch/index/shard/IndexShard.java:1105]: Did not appear to be called outside of the `_upgrade` API.

**Review by dakrone — APPROVED** [review]
LGTM, nice stats :)

**Pull request merged**